BootLoader.loadClassOrNull() NPE during native tests execution with JDK 11

The following exception is thrown during the `quarkus-integration-test-spring-data-jpa` native tests execution:
```
2019-12-12 01:09:59,126 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-worker-thread-1) HTTP Request to /post/all failed, error id: 3a612a18-7a5a-4c82-b732-4c596e1dcc39-2: org.jboss.resteasy.spi.UnhandledException: javax.ws.rs.ProcessingException: RESTEASY008205: JSON Binding serialization error javax.json.bind.JsonbException: Internal error: null
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:381)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:209)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeResponse(SynchronousDispatcher.java:587)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:508)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:252)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:153)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:363)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:156)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:238)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:114)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.lambda$handle$0(VertxRequestHandler.java:79)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:316)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:834)
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:479)
        at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: javax.ws.rs.ProcessingException: RESTEASY008205: JSON Binding serialization error javax.json.bind.JsonbException: Internal error: null
        at org.jboss.resteasy.plugins.providers.jsonb.JsonBindingProvider.writeTo(JsonBindingProvider.java:141)
        at org.jboss.resteasy.core.interception.jaxrs.AbstractWriterInterceptorContext.writeTo(AbstractWriterInterceptorContext.java:193)
        at org.jboss.resteasy.core.interception.jaxrs.ServerWriterInterceptorContext.writeTo(ServerWriterInterceptorContext.java:64)
        at org.jboss.resteasy.core.interception.jaxrs.AbstractWriterInterceptorContext.proceed(AbstractWriterInterceptorContext.java:155)
        at org.jboss.resteasy.core.ServerResponseWriter.lambda$writeNomapResponse$2(ServerResponseWriter.java:156)
        at org.jboss.resteasy.core.interception.jaxrs.ContainerResponseContextImpl.filter(ContainerResponseContextImpl.java:404)
        at org.jboss.resteasy.core.ServerResponseWriter.executeFilters(ServerResponseWriter.java:232)
        at org.jboss.resteasy.core.ServerResponseWriter.writeNomapResponse(ServerResponseWriter.java:97)
        at org.jboss.resteasy.core.ServerResponseWriter.writeNomapResponse(ServerResponseWriter.java:70)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeResponse(SynchronousDispatcher.java:578)
        ... 16 more
Caused by: javax.json.bind.JsonbException: Internal error: null
        at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:82)
        at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:102)
        at org.eclipse.yasson.internal.JsonBinding.toJson(JsonBinding.java:118)
        at org.jboss.resteasy.plugins.providers.jsonb.JsonBindingProvider.writeTo(JsonBindingProvider.java:137)
        ... 25 more
Caused by: java.lang.NullPointerException
        at jdk.internal.loader.BootLoader.loadClassOrNull(BootLoader.java:116)
        at jdk.internal.loader.BootLoader.loadClass(BootLoader.java:124)
        at java.lang.Package.getPackageInfo(Package.java:422)
        at java.lang.Package.getAnnotations(Package.java:467)
        at org.eclipse.yasson.internal.AnnotationIntrospector.collectAnnotations(AnnotationIntrospector.java:704)
        at org.eclipse.yasson.internal.MappingContext.lambda$createParseClassModelFunction$1(MappingContext.java:89)
        at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
        at org.eclipse.yasson.internal.MappingContext.getOrCreateClassModel(MappingContext.java:82)
        at org.eclipse.yasson.internal.serializer.AbstractContainerSerializer.serializeItem(AbstractContainerSerializer.java:145)
        at org.eclipse.yasson.internal.serializer.CollectionSerializer.serializeInternal(CollectionSerializer.java:39)
        at org.eclipse.yasson.internal.serializer.CollectionSerializer.serializeInternal(CollectionSerializer.java:27)
        at org.eclipse.yasson.internal.serializer.AbstractContainerSerializer.serialize(AbstractContainerSerializer.java:64)
        at org.eclipse.yasson.internal.Marshaller.serializeRoot(Marshaller.java:148)
        at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:76)
        ... 28 more
```
This issue is similar to #6089 as they have the same root cause: `ClassLoaders.bootLoader()` is `null` in native mode. They require different substitutions to be fixed though.

--- a/core/runtime/src/main/java/io/quarkus/runtime/graal/BootLoaderSubstitutions.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/graal/BootLoaderSubstitutions.java
@@ -7,17 +7,30 @@ import java.util.Enumeration;
 
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.hub.ClassForNameSupport;
 import com.oracle.svm.core.jdk.JDK11OrLater;
 
-/*
- * The `hasClassPath()` method substitution from this class is required to work around a NPE when `BootLoader.hasClassPath()`
- * is called. This was fixed in a GraalVM commit (https://github.com/oracle/graal/commit/68027de) which will be backported to
- * 19.3.1. We copied the entire GraalVM commit content into Quarkus for the sake of consistency.
- * See https://github.com/oracle/graal/issues/1966 for more details about the NPE.
- */
 @TargetClass(className = "jdk.internal.loader.BootLoader", onlyWith = JDK11OrLater.class)
 final class Target_jdk_internal_loader_BootLoader {
 
+    /*
+     * The following substitution is required to work around a NPE that happened when `BootLoader.loadClassOrNull(name)` was
+     * called during the Quarkus native integration tests execution. It will be directly fixed in a future GraalVM release
+     * which is undetermined for now. This substitution should be removed as soon as Quarkus depends on a GraalVM version that
+     * includes the fix.
+     */
+    @Substitute
+    private static Class<?> loadClassOrNull(String name) {
+        return ClassForNameSupport.forNameOrNull(name, false);
+    }
+
+    /*
+     * The following substitution is required to work around a NPE that happened when `BootLoader.hasClassPath()` was called
+     * during the Quarkus native integration tests execution. It was fixed in a GraalVM commit which should be backported to
+     * 19.3.1 (https://github.com/oracle/graal/commit/68027de). This substitution should be removed as soon as Quarkus depends
+     * on a GraalVM version that includes that commit.
+     * See https://github.com/oracle/graal/issues/1966 for more details about the NPE.
+     */
     @Substitute
     private static boolean hasClassPath() {
         return true;

